Installation issue: SFCGAL

### Steps to reproduce the issue

```console
% spack spec -l sfcgal

Input spec
--------------------------------
 -   sfcgal

Concretized
--------------------------------
 -   24tvyjg  sfcgal@1.3.8%apple-clang@15.0.0~ipo build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m2
[+]  oztywyi      ^boost@1.83.0%apple-clang@15.0.0~atomic+chrono~clanglibcpp+container~context~contract~coroutine~date_time~debug+exception~fiber+filesystem~graph~graph_parallel~icu~iostreams~json~locale~log+math~mpi+multithreaded~nowide~numpy~pic+program_options~python+random~regex+serialization+shared~signals~singlethreaded~stacktrace+system~taggedlayout+test+thread+timer~type_erasure~versionedlayout~wave build_system=generic cxxstd=98 patches=a440f96 visibility=hidden arch=darwin-sonoma-m2
[+]  hw3gk3k      ^cgal@5.5.2%apple-clang@15.0.0+core~demos+eigen~header_only~imageio~ipo+shared build_system=cmake build_type=Release generator=make arch=darwin-sonoma-m2
[+]  vnt4fes          ^eigen@3.4.0%apple-clang@15.0.0~ipo build_system=cmake build_type=RelWithDebInfo generator=make arch=darwin-sonoma-m2
[+]  pvkio5y          ^zlib-ng@2.1.5%apple-clang@15.0.0+compat+opt build_system=autotools arch=darwin-sonoma-m2
[+]  o4yw3z5      ^cmake@3.27.9%apple-clang@15.0.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=darwin-sonoma-m2
[+]  es6ufgq          ^curl@8.4.0%apple-clang@15.0.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=secure_transport arch=darwin-sonoma-m2
[+]  vna5isy              ^nghttp2@1.57.0%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  5hbreiv              ^pkgconf@1.9.5%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  nz36btw          ^ncurses@6.4%apple-clang@15.0.0~symlinks+termlib abi=none build_system=autotools arch=darwin-sonoma-m2
[+]  qumc4hw      ^gmake@4.4.1%apple-clang@15.0.0~guile build_system=generic arch=darwin-sonoma-m2
[+]  2tc5utg      ^gmp@6.2.1%apple-clang@15.0.0+cxx build_system=autotools libs=shared,static patches=69ad2e2 arch=darwin-sonoma-m2
[+]  tt6h5br          ^autoconf@2.69%apple-clang@15.0.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=darwin-sonoma-m2
[+]  d2tmw4b              ^perl@5.38.0%apple-clang@15.0.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=darwin-sonoma-m2
[+]  zzhn6bq                  ^berkeley-db@18.1.40%apple-clang@15.0.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=darwin-sonoma-m2
[+]  vqyl5ow                  ^bzip2@1.0.8%apple-clang@15.0.0~debug~pic+shared build_system=generic arch=darwin-sonoma-m2
[+]  suam277                  ^gdbm@1.23%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  ezbg36i                      ^readline@8.2%apple-clang@15.0.0 build_system=autotools patches=bbf97f1 arch=darwin-sonoma-m2
[+]  px5vslc          ^automake@1.16.5%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  ybxeibk          ^gnuconfig@2022-09-17%apple-clang@15.0.0 build_system=generic arch=darwin-sonoma-m2
[+]  ywmnfbj          ^libtool@2.4.7%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  4opbggi          ^m4@1.4.19%apple-clang@15.0.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=darwin-sonoma-m2
[+]  om7om6o              ^diffutils@3.9%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  64wjeqt                  ^libiconv@1.17%apple-clang@15.0.0 build_system=autotools libs=shared,static arch=darwin-sonoma-m2
[+]  7ci6kam              ^libsigsegv@2.14%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  essgvkl      ^mpfr@4.2.0%apple-clang@15.0.0 build_system=autotools libs=shared,static arch=darwin-sonoma-m2
[+]  cb5dmpg          ^autoconf-archive@2023.02.20%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  6wuvy4e          ^texinfo@7.0.3%apple-clang@15.0.0 build_system=autotools arch=darwin-sonoma-m2
[+]  urcpst6              ^gettext@0.22.4%apple-clang@15.0.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=darwin-sonoma-m2
[+]  yoaeqlu                  ^libxml2@2.10.3%apple-clang@15.0.0+pic~python+shared build_system=autotools arch=darwin-sonoma-m2
[+]  lvk7xip                  ^tar@1.34%apple-clang@15.0.0 build_system=autotools zip=pigz arch=darwin-sonoma-m2
[+]  nz2sfpy                      ^pigz@2.7%apple-clang@15.0.0 build_system=makefile arch=darwin-sonoma-m2
[+]  vbox53f                      ^zstd@1.5.5%apple-clang@15.0.0+programs build_system=makefile compression=none libs=shared,static arch=darwin-sonoma-m2
[+]  sxsxwvb                  ^xz@5.4.1%apple-clang@15.0.0~pic build_system=autotools libs=shared,static arch=darwin-sonoma-m2

...
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8'

8 errors found in build log:
     250    cd /private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-
            build-24tvyjg/src && /Users/johnnyamos/SRC/spack/lib/spack/env/clang/clang++ -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK
             -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DCGAL_USE_CORE=1 -DCGAL_
            USE_GMPXX=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sf
            cgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-build-24tvyjg/include -I/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../CGAL_patches -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma
            -m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-cl
            ang-15.0.0/boost-1.83.0-oztywyi6tos3srpebbjpnof2fb3j5mvd/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0
            /gmp-6.2.1-2tc5utgfvolgdcwvbdofb4qbfgk6vr3q/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/mpfr-4.2.0-e
            ssgvkl2ycd37wxpfcfxbvz5k2ycim2d/include -O3 -DNDEBUG -std=gnu++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX
            .platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -MD -MT src/CMakeFiles/SFCGAL.dir/algorithm/area.cpp.o -MF CMakeFiles/SFCGAL.di
            r/algorithm/area.cpp.o.d -o CMakeFiles/SFCGAL.dir/algorithm/area.cpp.o -c /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/algorithm/area.cpp
     251    [ 29%] Building CXX object src/CMakeFiles/SFCGAL.dir/algorithm/collect.cpp.o
     252    cd /private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-
            build-24tvyjg/src && /Users/johnnyamos/SRC/spack/lib/spack/env/clang/clang++ -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK
             -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DCGAL_USE_CORE=1 -DCGAL_
            USE_GMPXX=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sf
            cgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-build-24tvyjg/include -I/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../CGAL_patches -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma
            -m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-cl
            ang-15.0.0/boost-1.83.0-oztywyi6tos3srpebbjpnof2fb3j5mvd/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0
            /gmp-6.2.1-2tc5utgfvolgdcwvbdofb4qbfgk6vr3q/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/mpfr-4.2.0-e
            ssgvkl2ycd37wxpfcfxbvz5k2ycim2d/include -O3 -DNDEBUG -std=gnu++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX
            .platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -MD -MT src/CMakeFiles/SFCGAL.dir/algorithm/collect.cpp.o -MF CMakeFiles/SFCGAL
            .dir/algorithm/collect.cpp.o.d -o CMakeFiles/SFCGAL.dir/algorithm/collect.cpp.o -c /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spac
            k-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/algorithm/collect.cpp
     253    [ 30%] Building CXX object src/CMakeFiles/SFCGAL.dir/algorithm/collectionExtract.cpp.o
     254    cd /private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-
            build-24tvyjg/src && /Users/johnnyamos/SRC/spack/lib/spack/env/clang/clang++ -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK
             -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DCGAL_USE_CORE=1 -DCGAL_
            USE_GMPXX=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sf
            cgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-build-24tvyjg/include -I/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../CGAL_patches -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma
            -m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-cl
            ang-15.0.0/boost-1.83.0-oztywyi6tos3srpebbjpnof2fb3j5mvd/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0
            /gmp-6.2.1-2tc5utgfvolgdcwvbdofb4qbfgk6vr3q/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/mpfr-4.2.0-e
            ssgvkl2ycd37wxpfcfxbvz5k2ycim2d/include -O3 -DNDEBUG -std=gnu++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX
            .platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -MD -MT src/CMakeFiles/SFCGAL.dir/algorithm/collectionExtract.cpp.o -MF CMakeFi
            les/SFCGAL.dir/algorithm/collectionExtract.cpp.o.d -o CMakeFiles/SFCGAL.dir/algorithm/collectionExtract.cpp.o -c /var/folders/r3/05jyzbns7m5_s7dg2n
            qqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/algorithm/collectionExtract.cpp
     255    In file included from /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7
            bg4b/spack-src/src/algorithm/Intersection3D.cpp:24:
  >> 256    /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../
            CGAL_patches/CGAL/intersection_of_Polyhedra_3.h:1063:19: error: no member named 'Intersect_facets' in namespace 'CGAL::internal'
     257            internal::Intersect_facets<Polyhedron,Kernel,
     258            ~~~~~~~~~~^
  >> 259    /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../
            CGAL_patches/CGAL/intersection_of_Polyhedra_3.h:1063:36: error: 'Polyhedron' does not refer to a value
     260            internal::Intersect_facets<Polyhedron,Kernel,
     261                                       ^
     262    /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../
            CGAL_patches/CGAL/intersection_of_Polyhedra_3.h:795:17: note: declared here
     263    template< class Polyhedron,
     264                    ^
  >> 265    /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../
            CGAL_patches/CGAL/intersection_of_Polyhedra_3.h:1068:70: error: use of undeclared identifier 'intersect_facets'
     266            CGAL::box_self_intersection_d(box_ptr.begin(), box_ptr.end(),intersect_facets,cutoff);
     267                                                                         ^
  >> 268    /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../
            CGAL_patches/CGAL/intersection_of_Polyhedra_3.h:1071:14: error: no type named 'Throw_at_output_exception' in 'CGAL::internal::Throw_at_output'; did
             you mean 'internal::Throw_at_output_exception'?
     269          catch( internal::Throw_at_output::Throw_at_output_exception& )
     270                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     271                 internal::Throw_at_output_exception
     272    /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include/CGAL/exceptions.h:190
            :7: note: 'internal::Throw_at_output_exception' declared here
     273    class Throw_at_output_exception : public std::exception { };
     274          ^

     ...

     290    cd /private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-
            build-24tvyjg/src && /Users/johnnyamos/SRC/spack/lib/spack/env/clang/clang++ -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK
             -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DCGAL_USE_CORE=1 -DCGAL_
            USE_GMPXX=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sf
            cgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-build-24tvyjg/include -I/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../CGAL_patches -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma
            -m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-cl
            ang-15.0.0/boost-1.83.0-oztywyi6tos3srpebbjpnof2fb3j5mvd/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0
            /gmp-6.2.1-2tc5utgfvolgdcwvbdofb4qbfgk6vr3q/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/mpfr-4.2.0-e
            ssgvkl2ycd37wxpfcfxbvz5k2ycim2d/include -O3 -DNDEBUG -std=gnu++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX
            .platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -MD -MT src/CMakeFiles/SFCGAL.dir/algorithm/connection.cpp.o -MF CMakeFiles/SFC
            GAL.dir/algorithm/connection.cpp.o.d -o CMakeFiles/SFCGAL.dir/algorithm/connection.cpp.o -c /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnny
            amos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/algorithm/connection.cpp
     291    [ 34%] Building CXX object src/CMakeFiles/SFCGAL.dir/algorithm/convexHull.cpp.o
     292    cd /private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-
            build-24tvyjg/src && /Users/johnnyamos/SRC/spack/lib/spack/env/clang/clang++ -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK
             -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DCGAL_USE_CORE=1 -DCGAL_
            USE_GMPXX=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sf
            cgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-build-24tvyjg/include -I/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../CGAL_patches -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma
            -m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-cl
            ang-15.0.0/boost-1.83.0-oztywyi6tos3srpebbjpnof2fb3j5mvd/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0
            /gmp-6.2.1-2tc5utgfvolgdcwvbdofb4qbfgk6vr3q/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/mpfr-4.2.0-e
            ssgvkl2ycd37wxpfcfxbvz5k2ycim2d/include -O3 -DNDEBUG -std=gnu++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX
            .platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -MD -MT src/CMakeFiles/SFCGAL.dir/algorithm/convexHull.cpp.o -MF CMakeFiles/SFC
            GAL.dir/algorithm/convexHull.cpp.o.d -o CMakeFiles/SFCGAL.dir/algorithm/convexHull.cpp.o -c /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnny
            amos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/algorithm/convexHull.cpp
     293    In file included from /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7
            bg4b/spack-src/src/algorithm/Intersection3D.cpp:26:
     294    In file included from /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7
            bg4b/spack-src/src/../CGAL_patches/CGAL/corefinement_operations.h:30:
     295    In file included from /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7
            bg4b/spack-src/src/../CGAL_patches/CGAL/internal/corefinement/Combinatorial_map_output_builder.h:28:
  >> 296    /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../
            CGAL_patches/CGAL/internal/corefinement/Combinatorial_map_for_corefinement.h:29:10: fatal error: 'CGAL/Combinatorial_map_constructors.h' file not f
            ound
     297    #include <CGAL/Combinatorial_map_constructors.h>
     298             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     299    [ 35%] Building CXX object src/CMakeFiles/SFCGAL.dir/algorithm/covers.cpp.o
     300    cd /private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-
            build-24tvyjg/src && /Users/johnnyamos/SRC/spack/lib/spack/env/clang/clang++ -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK
             -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DBOOST_THREAD_DYN_LINK -DBOOST_THREAD_NO_LIB -DCGAL_USE_CORE=1 -DCGAL_
            USE_GMPXX=1 -DSFCGAL_BUILD_SHARED -DSFCGAL_EXPORTS -I/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sf
            cgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-build-24tvyjg/include -I/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/s
            pack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/../CGAL_patches -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma
            -m2/apple-clang-15.0.0/cgal-5.5.2-hw3gk3k27afiu5sf5awk555bb4yiaj2v/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-cl
            ang-15.0.0/boost-1.83.0-oztywyi6tos3srpebbjpnof2fb3j5mvd/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0
            /gmp-6.2.1-2tc5utgfvolgdcwvbdofb4qbfgk6vr3q/include -isystem /Users/johnnyamos/SRC/spack/opt/spack/darwin-sonoma-m2/apple-clang-15.0.0/mpfr-4.2.0-e
            ssgvkl2ycd37wxpfcfxbvz5k2ycim2d/include -O3 -DNDEBUG -std=gnu++14 -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX
            .platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=14.0 -fPIC -MD -MT src/CMakeFiles/SFCGAL.dir/algorithm/covers.cpp.o -MF CMakeFiles/SFCGAL.
            dir/algorithm/covers.cpp.o.d -o CMakeFiles/SFCGAL.dir/algorithm/covers.cpp.o -c /var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-s
            tage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r4k4jur2xjw3id7bg4b/spack-src/src/algorithm/covers.cpp
     301    1 warning and 5 errors generated.
  >> 302    make[2]: *** [src/CMakeFiles/SFCGAL.dir/build.make:401: src/CMakeFiles/SFCGAL.dir/algorithm/Intersection3D.cpp.o] Error 1
     303    make[2]: *** Waiting for unfinished jobs....
     304    make[2]: Leaving directory '/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r
            4k4jur2xjw3id7bg4b/spack-build-24tvyjg'
  >> 305    make[1]: *** [CMakeFiles/Makefile2:164: src/CMakeFiles/SFCGAL.dir/all] Error 2
     306    make[1]: Leaving directory '/private/var/folders/r3/05jyzbns7m5_s7dg2nqqr1300000gn/T/johnnyamos/spack-stage/spack-stage-sfcgal-1.3.8-24tvyjgofeyk7r
            4k4jur2xjw3id7bg4b/spack-build-24tvyjg'
  >> 307    make: *** [Makefile:149: all] Error 2

...
</pre></details>


### Information on your system

* **Spack:** 0.22.0.dev0 (5d999d0e4fb59b8eedc7519a6ed7d9361ce60688)
* **Python:** 3.9.6
* **Platform:** darwin-sonoma-m2
* **Concretizer:** clingo

### Additional information

I couldn't submit the issue with the files here because the data exceeded the maximum of 65536 characters. I will add the information to a reply.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/sfcgal/package.py
+++ b/var/spack/repos/builtin/packages/sfcgal/package.py
@@ -15,16 +15,37 @@ class Sfcgal(CMakePackage):
     """
 
     homepage = "http://www.sfcgal.org/"
-    url = "https://github.com/Oslandia/SFCGAL/archive/v1.3.8.tar.gz"
+    url = "https://gitlab.com/sfcgal/SFCGAL/-/archive/v1.5.1/SFCGAL-v1.5.1.tar.gz"
+    # URL for versions up to 1.3.8
+    old_github_urlbase = "https://github.com/Oslandia/SFCGAL/archive/v{0}.tar.gz"
 
     license("LGPL-2.0-or-later")
 
-    version("1.3.8", sha256="5154bfc67a5e99d95cb653d70d2b9d9293d3deb3c8f18b938a33d68fec488a6d")
-    version("1.3.7", sha256="30ea1af26cb2f572c628aae08dd1953d80a69d15e1cac225390904d91fce031b")
+    version("1.5.1", sha256="ea5d1662fada7de715ad564dc810c3059024ed81ae393f5352489f706fdfa3b1")
+    version("1.4.1", sha256="1800c8a26241588f11cddcf433049e9b9aea902e923414d2ecef33a3295626c3")
+    version(
+        "1.3.8",
+        sha256="5154bfc67a5e99d95cb653d70d2b9d9293d3deb3c8f18b938a33d68fec488a6d",
+        url=old_github_urlbase.format("1.3.8"),
+    )
+    version(
+        "1.3.7",
+        sha256="30ea1af26cb2f572c628aae08dd1953d80a69d15e1cac225390904d91fce031b",
+        url=old_github_urlbase.format("1.3.7"),
+    )
 
     depends_on("cmake@2.8.6:", type="build")
     # Ref: https://oslandia.github.io/SFCGAL/installation.html, but starts to work @4.7:
-    depends_on("cgal@4.7: +core")
+    # Ref: https://gitlab.com/sfcgal/SFCGAL/-/blob/v1.5.1/NEWS?ref_type=tags
+    # and looking at CMakeLists.txt find_package(CGAL) declaration
+    # for different versions (around line 70)
+    # @1.3.8:1.3.10 (from comments) cgal@4.3 is minimal, @4.13 recommended, @5 supported?
+    depends_on("cgal +core")
+    depends_on("cgal@4.7:4", when="@1.3.8")
+    depends_on("cgal@4.7:5.1", when="@1.3.9")
+    depends_on("cgal@4.7:5.2", when="@1.3.10")
+    depends_on("cgal@5.3", when="@1.4")
+    depends_on("cgal@5.6", when="@1.5")
     depends_on(
         "boost@1.54.0:+chrono+filesystem+program_options+serialization+system+test+thread+timer"
     )

